src/zope/interface/registry.py

```diff
@@ -184,6 +145,9 @@ def __init__(self, name='', bases=()):
         self._init_registrations()
         self.__bases__ = tuple(bases)
 
+        if hasattr(self, '_v_utility_registrations_cache'):
+            del self._v_utility_registrations_cache
```

Comment:
As a reader of this code, I see this and I think, "Why is that being done in `__init__`? There should be no possible way this object can have that attribute, it isn't set anywhere yet." The secret, of course, is that somewhere else---not even in this project!---there's code that registers a cleanup with zope.testing to call `__init__`. That's pretty non-obvious; most people don't consider `__init__` being called on existing objects multiple times to be "normal". How about just setting that attribute to `None`? (I know, that voids the `AttributeError` optimization in the property...but maybe that optimization is premature.) Or have the property have a setter that does the deletion; that makes a certain sense from an internal-logic standpoint.